synapse/servers/cell.py

```diff
@@ -0,0 +1,86 @@
+import os
+import sys
+import asyncio
+import logging
+import argparse
+
+import synapse.exc as s_exc
+import synapse.common as s_common
+
+import synapse.lib.base as s_base
+import synapse.lib.output as s_output
+import synapse.lib.dyndeps as s_dyndeps
+
+logger = logging.getLogger(__name__)
+
+
+async def getCell(outp,
+                  celldir,
+                  ctorpath,
+                  httpport,
+                  telepath,
+                  name=None,
+                  desc='cell'):
+    ctor = s_dyndeps.getDynLocal(ctorpath)
+    if ctor is None:
+        raise s_exc.NoSuchCtor(name=ctorpath,
+                               mesg='No Cell ctor found.')
+
+    outp.printf(f'starting {desc}: {celldir}')
+
+    cell = await ctor.anit(celldir)
+
+    try:
+
+        outp.printf(f'...{desc} API (telepath): {telepath}')
+        await cell.dmon.listen(telepath)
+
+        outp.printf(f'...{desc} API (https): {httpport}')
+        await cell.addHttpsPort(httpport)
+
+        if name:
+            outp.printf(f'...{desc} additional share name: {name}')
+            cell.dmon.share(name, cell)
+
+        return cell
+
+    except Exception:
+        await cell.fini()
+        raise
+
+
+def parse(argv):
+    cctor = os.getenv('SYN_UNIV_CELL', 'synapse.lib.cell.Cell')
+    https = os.getenv('SYN_UNIV_HTTPS', '4443')
+    telep = os.getenv('SYN_UNIV_TELEPATH', 'tcp://0.0.0.0:27492/')
+    telen = os.getenv('SYN_UNIV_NAME', None)
+
+    pars = argparse.ArgumentParser(prog='synapse.servers.cell',
+                                   description='A universal Synapse Cell loader.')
+    pars.add_argument('--cell', default=cctor, help='Python class path to use to load the Cell.')
```

Comment:
This argument should be mandatory i think.  Doesn't seem to make sense to execute without it.

Reply:
So it starts with a default value of `synapse.lib.cell.Cell` and is controlled via a envar.

Reply:
made the ctor required